### PR TITLE
Fix/new ui is distorted due to long organization name #735

### LINKED ISSUE #735: [BUG] New UI is distorted due to long organization name
Attachment: https://drive.google.com/drive/folders/1JzmFpTXIfmmO89cQRSAR9sA0caJG8kWN?usp=sharing

Steps to replicate:
1. Go to Appsmith homepage
2. Observe Organizations listing on the left hand side of the page

Browser Details: Firefox
Device Details: macOS 10.15.6 (MacBook Pro)
Note: Tried on cloud.

### PR DESCRIPTION
**Pull request by A-Scratchy**
## Description
Overly long organisation names were breaking the UI. 
Here we have implemented an optional ellipsize parameter that can be set when rendering a `MenuItem` component.
If the string is over the set limit, the string will be shortened and an ellipse added to the end. The original full text will be visible in a tooltip when the user hovers over the item.

*additional comments on the individual commits...*

Fixes #735 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

#### Manual testing
Tested manually and verified on the following systems.
- Firefox on debian stretch 10
- Chrome on debian stretch 10 

#### Cypress
Ran Cypress tests, however couldn't get them to work even before the changes, failing on API calls etc, must be to do with the local dev setup. No noticeable change in their behaviour after changes made.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
*required for this?*
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works 
 *Cannot get tests to work. Happy to work on this if someone can help me set them up.*
- [ ] New and existing unit tests pass locally with my changes
*as above*

### CODE REVIEW

**Comment by hetunandu** [comment]
Hey @A-Scratchy we had an issue with our CI runs which did not run the tests properly. Can you rebase/merge the latest release branch into your branch to fix this?

File: app/client/src/components/ads/MenuItem.tsx [hunk 1]
```diff
@@ -84,4 +100,8 @@ function MenuItem(props: MenuItemProps) {
   );
 }
 
+function ellipsize(length: number, text: string) {
+  return text.length > length ? text.slice(0, length).concat(" ...") : text
+};
```
Comment (Nikhil-Nandagopal) [inline app/client/src/components/ads/MenuItem.tsx:105]: @A-Scratchy can you also truncate the text on the main home page

Comment (A-Scratchy) [reply app/client/src/components/ads/MenuItem.tsx:105]: Is this ok now? I just checked out the branch to have a look. In the process I resolved the merge conflicts. Do you want me to push?

Comment (Nikhil-Nandagopal) [reply app/client/src/components/ads/MenuItem.tsx:105]: @A-Scratchy yes sounds good

**Comment by A-Scratchy** [comment]
@Nikhil-Nandagopal 
I have tidied up this pull request and merged with latest release branch.
The shortening of Org names in the main window and org submenu has been solved in #1010 
My PR now fixes the issue of shortening org names in left panel and adding tooltip on hover.
Thanks :-)

**Review by Nikhil-Nandagopal — APPROVED** [review]
LGTM

**Pull request merged**